Makefile

```diff
@@ -295,25 +319,34 @@ $(BUILT)/add-on/chrome/locale/%/loop.properties: locale/%/add-on.properties loca
 # specific rules, i.e., a target for a parent directory (which happens to match
 # all sub directories) comes before a sub directory with a different `PREREQ`.
 # This ordering can be achieved by keeping the list lexicographically sorted.
-$(BUILT)/add-on/chrome/content/panels/js/%.js: PREREQ = add-on/panels/js/$(@F)x
-$(BUILT)/add-on/chrome/content/panels/vendor/%.js: PREREQ = add-on/panels/vendor/$(@F)x
-$(BUILT)/add-on/chrome/content/shared/js/%.js: PREREQ = shared/js/$(@F)x
-$(BUILT)/standalone/content/js/%.js: PREREQ = standalone/content/js/$(@F)x
-$(BUILT)/standalone/content/shared/js/%.js: PREREQ = shared/js/$(@F)x
-$(BUILT)/ui/%.js: PREREQ = ui/$(@F)x
-$(BUILT)/ui/shared/js/%.js: PREREQ = shared/js/$(@F)x
-
-# During Makefile's secondary $$(variable) expansion, use the prerequisite
-# specified in the pattern-specific `PREREQ` to build one of the explicily
-# listed targets from the `built_*_js_files` variables.
+$(BUILT)/add-on/chrome/content/modules/%.jsm: PREREQ = add-on/chrome/modules/$(@F)
+$(BUILT)/add-on/chrome/content/modules/%.js: PREREQ = add-on/chrome/modules/$(@F)
+$(BUILT)/add-on/chrome/content/panels/js/%.js: PREREQ = add-on/panels/js/$(@F)*
+$(BUILT)/add-on/chrome/content/panels/test/%.js: PREREQ = add-on/panels/test/$(@F)
+$(BUILT)/add-on/chrome/content/panels/vendor/%.js: PREREQ = add-on/panels/vendor/$(@F)*
+$(BUILT)/add-on/chrome/content/shared/js/%.js: PREREQ = shared/js/$(@F)*
+$(BUILT)/add-on/chrome/content/shared/test/%.js: PREREQ = shared/test/$(@F)
+$(BUILT)/standalone/%.js: PREREQ = standalone/$(@F)
```

Comment:
Any specific reason this line is `$(@F)` instead of `$(@F)*` ? Might as well be consistent or add a comment why this is different.